### PR TITLE
feat: Template pubsublitetogcs java

### PR DESCRIPTION
**Pull request by saumyasinha-google**

### CODE REVIEW

File: java/.ci/Jenkinsfile [hunk 1]
```diff
@@ -769,7 +769,29 @@ pipeline {
...
             }
-            
+            stage('Parallel Execution 9'){
+                parallel{
+                    stage('PubSubLite To GCS') {
+                        steps{
+                            sh '''
+                                export GCP_PROJECT=yadavaja-sandbox
+                                export REGION=us-west1
+                                export GCS_STAGING_LOCATION="gs://python-dataproc-templates-temp"
```
Comment (vanshaj-bhatia) [inline java/.ci/Jenkinsfile:779]: These are not required.

File: java/.ci/Jenkinsfile
Comment (vanshaj-bhatia) [inline java/.ci/Jenkinsfile:1]: Add export SKIP_BUILD=true

File: java/.ci/Jenkinsfile [hunk 2]
```diff
@@ -769,7 +769,29 @@ pipeline {
...
+
+                                cd java
+
+                                bin/start.sh \
+                                -- --template PUBSUBLITETOGCS \
+                                --templateProperty pubsublite.to.gcs.input.subscription.url=projects/$GCP_PROJECT/locations/$REGION/subscriptions/psltobt-sub \
```
Comment (vanshaj-bhatia) [inline java/.ci/Jenkinsfile:785]: Will $GCP_PROJECT work? Or is the $GCP_PROJECT_NUMBER is required?
@balajiss2

File: java/.ci/Jenkinsfile [hunk 3]
```diff
@@ -769,7 +769,29 @@ pipeline {
...
+                                --templateProperty pubsublite.to.gcs.processing.time.seconds=1 \
+                                --templateProperty pubsublite.to.gcs.timeout.ms=120000 \
+                                --templateProperty pubsublite.to.gcs.output.location=gs://saumyasinha/pubsublite/output \
+                                --templateProperty pubsublite.to.gcs.checkpoint.location=gs://saumyasinha/pubsublite/checkpointLocation
```
Comment (vanshaj-bhatia) [inline java/.ci/Jenkinsfile:789]: Avoid using your bucket. As a best practice, please use a generic output location similar to other template's int test.

File: java/src/main/java/com/google/cloud/dataproc/templates/pubsublite/README.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    	--pubsublite.to.gcs.processing.time.seconds PUBSUBLITE.GCS.PROCESSING.TIME.SECONDS \
+
+optional arguments:
+  -h, --help            show this help message and exit
+  --pubsublite.to.gcs.write.mode PUBSUBLITE.TO.GCS.WRITE.MODE 
+            {overwrite,append,ignore,errorifexists} Output Write Mode (Defaults to append)
+  --pubsublite.to.gcs.output.format PUBSUBLITE.TO.GCS.OUTPUT.FORMAT
+            {avro,parquet,csv,json} Output Format (Defaults to json)
+```
+
```
Comment (vanshaj-bhatia) [inline java/src/main/java/com/google/cloud/dataproc/templates/pubsublite/README.md:35]: Why is python usage referenced here?

**Review by vanshaj-bhatia — CHANGES_REQUESTED** [review]

**Review by vanshaj-bhatia — CHANGES_REQUESTED** [review]

File: java/src/main/java/com/google/cloud/dataproc/templates/pubsublite/PubSubLiteToGCS.java [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+    // Set log level
+    spark.sparkContext().setLogLevel(sparkLogLevel);
+    Dataset<Row> df =
+        spark
+            .readStream()
+            .format("pubsublite")
+            .option("pubsublite.subscription", inputSubscriptionUrl)
+            .load();
+
```
[the remark this answers is not in the document]
Comment (saumyasinha-google) [reply java/src/main/java/com/google/cloud/dataproc/templates/pubsublite/PubSubLiteToGCS.java:76]: Done.

File: java/src/main/java/com/google/cloud/dataproc/templates/pubsublite/README.md (hunk 4 above)
Comment (saumyasinha-google) [reply java/src/main/java/com/google/cloud/dataproc/templates/pubsublite/README.md]: Removed. Added by mistake

File: java/.ci/Jenkinsfile (hunk 2 above)
Comment (balajiss2) [reply java/.ci/Jenkinsfile]: The $GCP_PROJECT_NUMBER is required. The $GCP_PROJECT did not work. I tried that already

File: java/.ci/Jenkinsfile (hunk 1 above)
Comment (saumyasinha-google) [reply java/.ci/Jenkinsfile]: Done.

File: java/.ci/Jenkinsfile (hunk 2 above)
Comment (saumyasinha-google) [reply java/.ci/Jenkinsfile]: It is working fine though with $GCP_PROJECT. Why do we need to change it, in python template also $GCP_PROJECT was used

File: java/.ci/Jenkinsfile
Comment (vanshaj-bhatia) [inline java/.ci/Jenkinsfile:1]: Please add a gcloud command to publish message to the lite topic "psltobt-sub". This has been done in python please refer the same.

File: java/src/main/java/com/google/cloud/dataproc/templates/util/TemplateConstants.java [hunk 6]
```diff
@@ -500,4 +500,14 @@ public interface TemplateConstants {
...
+
+  /** PubSubLite To GCS Properties */
+  String PUBSUBLITE_TO_GCS_INPUT_SUBSCRIPTION_URL = "pubsublite.to.gcs.input.subscription.url";
+
+  String PUBSUBLITE_TO_GCS_TIMEOUT_MS = "pubsublite.to.gcs.timeout.ms";
+  String PUBSUBLITE_TO_GCS_PROCESSING_TIME_SECONDS = "pubsublite.to.gcs.processing.time.seconds";
+  String PUBSUBLITE_TO_GCS_WRITE_MODE = "pubsublite.to.gcs.write.mode";
+  String PUBSUBLITE_TO_GCS_OUTPUT_FORMAT = "pubsublite.to.gcs.output.format";
+  String PUBSUBLITE_TO_GCS_OUTPUT_LOCATION = "pubsublite.to.gcs.output.location";
+  String PUBSUBLITE_CHECKPOINT_LOCATION = "pubsublite.to.gcs.checkpoint.location";
```
Comment (vanshaj-bhatia) [inline java/src/main/java/com/google/cloud/dataproc/templates/util/TemplateConstants.java:512]: Please rename it to PUBSUBLITE_TO_GCS_CHECKPOINT_LOCATION

**Review by vanshaj-bhatia — CHANGES_REQUESTED** [review]

File: java/src/main/resources/template.properties [hunk 7]
```diff
@@ -459,3 +459,11 @@ text.bigquery.output.dataset=
 text.bigquery.output.table=
 text.bigquery.output.mode=
 text.bigquery.temp.bucket=
+
+# PubSubLite To GCS
+pubsublite.to.gcs.input.subscription.url=
+pubsublite.to.gcs.processing.time.seconds=
```
Comment (tanyarw) [inline java/src/main/resources/template.properties:465]: processingTime can be optional with a default value here

File: java/src/main/resources/template.properties [hunk 8]
```diff
@@ -459,3 +459,11 @@ text.bigquery.output.dataset=
...
+pubsublite.to.gcs.write.mode=append
+pubsublite.to.gcs.output.format=json
+pubsublite.to.gcs.output.location=
+pubsublite.to.gcs.checkpoint.location
```
Comment (tanyarw) [inline java/src/main/resources/template.properties:469]: `pubsublite.to.gcs.timeout.ms` is missing. This parameter can be optional too, with a default value here

**Review by tanyarw — CHANGES_REQUESTED** [review]
Hey @saumyasinha-google, left a few comments & suggestions!

**Pull request closed without merging**